### PR TITLE
Fix z & m coordinates.

### PR DESCRIPTION
**Pull request by ghost**
Replace: https://github.com/qgis/QGIS/pull/5239

### CODE REVIEW

**Comment by haubourg** [comment]
@pblottiere can you review that for a merge ?

File: src/app/nodetool/qgsnodeeditor.cpp [hunk 1]
```diff
@@ -199,15 +199,16 @@ bool QgsNodeEditorModel::setData( const QModelIndex &index, const QVariant &valu
     return false;
   }
 
-  double x = ( index.column() == 0 ? value.toDouble() : mSelectedFeature->vertexMap().at( index.row() )->point().x() );
-  double y = ( index.column() == 1 ? value.toDouble() : mSelectedFeature->vertexMap().at( index.row() )->point().y() );
+  double doubleValue = QLocale().toDouble( value.toString() );
+
+  double x = ( index.column() == 0 ? doubleValue : mSelectedFeature->vertexMap().at( index.row() )->point().x() );
+  double y = ( index.column() == 1 ? doubleValue : mSelectedFeature->vertexMap().at( index.row() )->point().y() );
```
Comment (pblottiere) [inline src/app/nodetool/qgsnodeeditor.cpp]: It seems that those are `const` values.

File: src/core/qgsvectorlayereditutils.h [hunk 2]
```diff
@@ -189,6 +189,12 @@ class CORE_EXPORT QgsVectorLayerEditUtils
      */
     bool boundingBoxFromPointList( const QList<QgsPointXY> &list, double &xmin, double &ymin, double &xmax, double &ymax ) const;
 
+    /**
+     * Return default Z value
+     * Use for set Z coordinate to edited vertex for 2.5d geometries
+     */
```
Comment (pblottiere) [inline src/core/qgsvectorlayereditutils.h]: Can you add `\since QGIS 3.0` and take a look on typo (like `to edited` -> `to edit`)?

**Comment by pblottiere** [comment]
@nyalldawson You added the tag `Requires Changes` one week ago. What did you had in mind?

**Comment by nyalldawson** [comment]
Hm - I can't recall. It may have been an accident as I was tagging everything with outstanding comments as that, but in this case I can't see what the comment would have been...

**Comment by pblottiere** [comment]
Z/M already fixed by https://github.com/qgis/QGIS/pull/5571.

As discussed with @haubourg, this PR may be closed (we'll see later to move Z default value to project properties).

**Pull request closed without merging**